node/node.go

```diff
@@ -704,7 +704,10 @@ func (n *nodeImpl) OnStart() error {
 			if err != nil {
 				n.Logger.Error("state sync failed", "err", err)
 				// stop the node
-				n.Stop()
+				err := n.Stop()
+				if err != nil {
```

Comment:
I tend to combine these lines (also the err is already declared)

Reply:
Agreed, though this is the first declaration in this scope so it actually works (we shouldn't lean on that).